pkg/operator/staticpod/controllers.go

```diff
@@ -138,6 +151,21 @@ func (b *staticPodOperatorControllerBuilder) WithPruning(command []string, stati
 	return b
 }
 
+func (b *staticPodOperatorControllerBuilder) WithGuardDeployment(assetFunc func(string) []byte, file string) Builder {
```

Comment:
The guard deployment and PDB are both specified together, right?  this is the point to combine the two.

`WithPDB(deploymentAssetFn, pdbAssetFn, etc)`